buildSrc/src/main/kotlin/com/bmuschko/gradle/docker/PublishingPlugin.kt

```diff
@@ -0,0 +1,137 @@
+package com.bmuschko.gradle.docker
+
+import com.jfrog.bintray.gradle.BintrayExtension
+import com.jfrog.bintray.gradle.BintrayPlugin
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.kotlin.dsl.*
+import java.text.SimpleDateFormat
+import java.util.*
+
+class PublishingPlugin : Plugin<Project> {
+    override fun apply(project: Project): Unit = project.run {
+        applyPublishingPlugin()
+        configurePublishingExtension()
+        configureBintrayExtension()
+    }
+
+    private
+    fun Project.applyPublishingPlugin() {
+        plugins.apply(MavenPublishPlugin::class)
+        plugins.apply(BintrayPlugin::class)
+    }
+
+    private
+    fun Project.configurePublishingExtension() {
+        val sourcesJar: Jar by tasks.getting
+        val groovydocJar: Jar by tasks.getting
+        val javadocJar: Jar by tasks.getting
+
+        configure<PublishingExtension> {
+            publications {
+                create("mavenJava", MavenPublication::class) {
```

Comment:
Can be simplified to `create<MavenPublication>("mavenJava") {`

Reply:
Thanks, got this from https://github.com/gradle/kotlin-dsl/blob/master/samples/maven-publish/build.gradle.kts#L32. You may want to change it there as well.